### ISSUE TITLE
ffmpeg error when generating sprites and previews

### ISSUE THREAD

**Issue by rudi123github** [stashapp/stash#48]
When I generate previews and spites, I get this kind of error on some vids:

ERRO[3587] ffmpeg error when running command </usr/bin/ffmpeg -v quiet -ss 1180 -t 0.75 -i /home/user/Desktop/sort/Goga-Huxley ++.wmv -y -c:v libx264 -pix_fmt yuv420p -profile:v high -level 4.2 -preset veryslow -crf 21 -threads 4 -vf scale=640:-2 -c:a aac -b:a 128k -strict -2 /home/ziggy/stash/generated/tmp/preview010.mp4>

I ran the command manually from bash, and it does work if I add the option "-max_muxing_queue_size 400" to ffmpeg.

**Comment by rudi123github** [comment]
Ok, I figured out that the change needs to happen in pkg/ffmpeg/encoder_scene_preview_chunk.go. I added the same parameter I found in another module as extra argument to function ScenePreviewVideoChunk:

"-max_muxing_queue_size", "1024", // https://trac.ffmpeg.org/ticket/6375

I rebuilt stash with the docker container and now everything runs fine locally. I cannot push the update to github, so I ask one of the devs to make this change please.

**Comment by Leopere** [comment]
@rudi123github did you still get this issue or can it be closed?

**Comment by rudi123github** [comment]
I fixed this in my local build but the change should go into the development branch.

**Comment by Leopere** [comment]
If you file a PR it can get reviewed / worked on.  As far as I'm aware this app is far from 1.0 status but whatever works.

**Comment by rudi123github** [comment]
Okay, I just figured out how to do that. The PR is there now.

**Comment by ExceptionalError** [comment]
I am still getting errors when trying to generate preview images.
Since my filename contains spaces, I am not sure if it should be quoted.

Can anyone confirm it works with spaces?
Also the ````-max_muxing_queue_size 1024```` is missing.
````
msg="ffmpeg error when running command </usr/bin/ffmpeg -v quiet -ss 1562 -y -i /data/MyVideo [x264].mp4 -vframes 1 -q:v 1 -vf scale=160:-1 -f image2 /generated/tmp/thumbnail071.jpg>"
````

**Comment by rudi123github** [comment]
Hello, this is a different function. Can you please run this from a terminal:

 /usr/bin/ffmpeg -ss 1562 -y -i "/data/MyVideo [x264].mp4" -vframes 1 -q:v 1 -vf scale=160:-1 -f image2 /generated/tmp/thumbnail071.jpg

and post the output here?

**Comment by ExceptionalError** [comment]
If I encapsulate the filename with ```"``` it works, otherwise I get errors.

**Comment by ExceptionalError** [comment]
@Leopere : I pulled the latest docker image but I am still getting the error above? Could it be that my PR ist not included in the docker build?

**Comment by Leopere** [comment]
@ExceptionalError looks like everything I merged was built I wonder why its not working as prescribed.   For good measure I've rebuilt it one more time just in case it didn't work properly the last time but I doubt that anything will change.

**Comment by ExceptionalError** [comment]
Something changed, there are now quotes but somehow they get escaped:
````
/usr/bin/ffmpeg -v quiet -ss 2320 -y -i \"/data/MyVideo [x264].mp4\" -vframes 1 -q:v 1 -vf scale=160:-1 -f image2 /generated/tmp/thumbnail080.jpg
````

**Comment by ExceptionalError** [comment]
I have tried some more fixes in  #131 - what do you think?

**Comment by rudi123github** [comment]
I wonder if this is really related to spaces. I have many files with spaces in their names and all of these work without problems (I am on Linux). I think it must be a different character. Does your fixes version work for you?

**Comment by ExceptionalError** [comment]
> I think it must be a different character

Could you try renaming one of your files like this ````[MyTag] MyFilename (09.10.2019) [x264].mkv```` and see if you get errors, too?

**Comment by bnkai** [comment]
@ExceptionalError on linux at least spaces and your naming scheme works fine. The quoted part changes you did to 117 though broke that though.  #131 breaks the screenshot generation in the linux build also

**Comment by jpegsarebad** [comment]
Having this error with the most recent linux build, is this a different issue or is this still unfixed?
`ERRO[0354] ffmpeg error when running command </usr/bin/ffmpeg -v quiet -ss 49.612 -y -i /home/jack/Applications/stash/videos/vid.mp4 -vframes 1 -q:v 5 -vf scale=320:-1 -f image2 screenshots/650b216cccc90fdb9bfeeab412c6c06c.thumb.jpg>`
`ERRO[0354] ffmpeg error when running command </usr/bin/ffmpeg -v quiet -ss 49.612 -y -i /home/jack/Applications/stash/videos/vid.mp4 -vframes 1 -q:v 2 -vf scale=1280:-1 -f image2 screenshots/650b216cccc90fdb9bfeeab412c6c06c.jpg> `

**Comment by ExceptionalError** [comment]
@bnkai : Thanks for the feedback. Any idea how to proceed?
When executing manually in Docker (with quotes) I can generate the previews successfully.
I changed #131 so it outputs the ffmpeg error message, too - maybe that's a start.

**Comment by bnkai** [comment]
@jpegsarebad you can try executing the command manually to see if that produces an error
e.g.

`usr/bin/ffmpeg -v quiet -ss 49.612 -y -i /home/jack/Applications/stash/videos/vid.mp4 -vframes 1 -q:v 5 -vf scale=320:-1 -f image2 thumb.jpg`

doing that at least we can get a hint if something's wrong with the video file or with the invocation of ffmpeg through stash

@ExceptionalError is the error for all the files or just some specific? you say you used docker, if you use the linux binary then instead of docker do you get the same errors? i haven't used the docker but as i said the linux binary doesn't have any problem with spaces or special characters like (  [ + in the filename

**Comment by jpegsarebad** [comment]
@bnkai Yes, running the command manually works and produces the image correctly.

**Comment by ExceptionalError** [comment]
@bnkai I get the error message for all my files. When executing the command in the docker image manually, it runs successfully.
Imho #131 needs to be merged so there is at least the error message in the log.

**Comment by jpegsarebad** [comment]
My problem was unrelated, the error was ffmpeg not being able to find the file due to an error in my video folder path.

**Comment by Leopere** [comment]
@ExceptionalError #131 is just waiting on confirmation from @WithoutPants and @bnkai to verify that there's nothing broken in the PR.

**Comment by bnkai** [comment]
@ExceptionalError if the error is for all the files have you checked that the folders you use for the metadata files have the right permissions or access rights ?
For #131 a simple extra print doesn't break anything but have you tried that ? i don't have an easy way to force an error just to check that....

**Comment by bnkai** [comment]
@jpegsarebad the only think  i can think off is also maybe there might be someting wrong with the permissions. have you checked that all the metadata directories are ok?
afaik the sprite images are created in the tmp directory and then concatenated to a single image file in the meta vtt folder

**Comment by ExceptionalError** [comment]
Whilst the log output it still not in the current docker build, I mounted the "/generated" folder with the docker compose file and everything seems to be working fine now.
Maybe the folder structure inside the container is not properly created?

**Comment by Leopere** [comment]
Since we merged that PR I'm going to close this but feel free to continue the conversation or start a new issue if theres any further discussion to be had.
